### ISSUE TITLE
No data pulled on dashboard

### ISSUE THREAD

**Issue by robertmachina** [Netflix/security_monkey#817]
Hi,

I just installed security monkey following "quick start guide" strictly. However, after log in the Web UI, I cannot see anything on it.

In /var/log/security_monkey/security_monkey.log, there are some errors I cannot understand. Could anyone help tell me what the problem is? Thanks.

Regards,
Robert
```
2017-09-07 10:00:20,838 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:48:31,210 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:50:04,880 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:52:35,844 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:54:08,582 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:56:17,824 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:56:18,327 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-09-08 01:56:18,832 INFO: Adding job tentatively -- it will be properly scheduled when the scheduler starts [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:287]
2017-09-08 01:56:18,833 INFO: Added job "_clear_old_exceptions (trigger: cron[hour='3', minute='0'], next run at: 2017-09-08 03:00:00)" to job store "default" [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:262]
2017-09-08 01:56:18,833 INFO: Scheduler started [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:574]
2017-09-08 03:00:00,000 INFO: Running job "_clear_old_exceptions (trigger: cron[hour='3', minute='0'], next run at: 2017-09-09 03:00:00)" (scheduled at 2017-09-08 03:00:00) [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:509]
2017-09-08 03:00:00,591 INFO: Job "_clear_old_exceptions (trigger: cron[hour='3', minute='0'], next run at: 2017-09-09 03:00:00)" executed successfully [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:527]
2017-09-08 10:06:40,745 ERROR: Exception on /login [POST] [in /usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py:1423]
Traceback (most recent call last):
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_RESTful-0.3.3-py2.7.egg/flask_restful/__init__.py", line 270, in error_router
    return original_handler(e)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_RESTful-0.3.3-py2.7.egg/flask_restful/__init__.py", line 270, in error_router
    return original_handler(e)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/decorators.py", line 230, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/views.py", line 76, in login
    if form.validate_on_submit():
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_WTF-0.14.2-py2.7.egg/flask_wtf/form.py", line 101, in validate_on_submit
    return self.is_submitted() and self.validate()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/forms.py", line 239, in validate
    self.user = _datastore.get_user(self.email.data)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/datastore.py", line 191, in get_user
    rv = self.user_model.query.filter(query).first()
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2320, in first
    ret = list(self[0:1])
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2187, in __getitem__
    return list(res)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2391, in __iter__
    return self._execute_and_instances(context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2406, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 717, in execute
    return meth(self, multiparams, params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/sql/elements.py", line 317, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 814, in _execute_clauseelement
    compiled_sql, distilled_params
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 927, in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1076, in _handle_dbapi_exception
    exc_info
  File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 185, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 920, in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 425, in do_execute
    cursor.execute(statement, parameters)
OperationalError: (OperationalError) terminating connection due to administrator command
SSL connection has been closed unexpectedly
 'SELECT "user".id AS user_id, "user".email AS user_email, "user".password AS user_password, "user".active AS user_active, "user".confirmed_at AS user_confirmed_at, "user".daily_audit_email AS user_daily_audit_email, "user".change_reports AS user_change_reports, "user".last_login_at AS user_last_login_at, "user".current_login_at AS user_current_login_at, "user".login_count AS user_login_count, "user".last_login_ip AS user_last_login_ip, "user".current_login_ip AS user_current_login_ip, "user".role AS user_role \nFROM "user" \nWHERE "user".email ILIKE %(email_1)s \n LIMIT %(param_1)s' {'email_1': u'bob.ma@nin.com', 'param_1': 1}
2017-09-08 10:07:15,520 ERROR: Internal Error [in /usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_RESTful-0.3.3-py2.7.egg/flask_restful/__init__.py:299]
Traceback (most recent call last):
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1473, in full_dispatch_request
    rv = self.preprocess_request()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/app.py", line 1666, in preprocess_request
    rv = func()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Principal-0.4.0-py2.7.egg/flask_principal.py", line 477, in _on_before_request
    identity = loader()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/core.py", line 232, in _identity_loader
    if not isinstance(current_user._get_current_object(), AnonymousUserMixin):
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Werkzeug-0.12.2-py2.7.egg/werkzeug/local.py", line 306, in _get_current_object
    return self.__local()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Login-0.4.0-py2.7.egg/flask_login/utils.py", line 26, in <lambda>
    current_user = LocalProxy(lambda: _get_user())
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Login-0.4.0-py2.7.egg/flask_login/utils.py", line 302, in _get_user
    current_app.login_manager._load_user()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Login-0.4.0-py2.7.egg/flask_login/login_manager.py", line 317, in _load_user
    return self.reload_user()
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Login-0.4.0-py2.7.egg/flask_login/login_manager.py", line 279, in reload_user
    user = self.user_callback(user_id)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/core.py", line 208, in _user_loader
    return _security.datastore.find_user(id=user_id)
  File "/usr/local/src/security_monkey/venv/lib/python2.7/site-packages/Flask_Security_Fork-2.0.1-py2.7.egg/flask_security/datastore.py", line 203, in find_user
    return self.user_model.query.filter_by(**kwargs).first()
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2320, in first
    ret = list(self[0:1])
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2187, in __getitem__
    return list(res)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2391, in __iter__
    return self._execute_and_instances(context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2406, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 717, in execute
    return meth(self, multiparams, params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/sql/elements.py", line 317, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 814, in _execute_clauseelement
    compiled_sql, distilled_params
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 927, in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1076, in _handle_dbapi_exception
    exc_info
  File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 185, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 920, in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 425, in do_execute
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 425, in do_execute
    cursor.execute(statement, parameters)
OperationalError: (OperationalError) terminating connection due to administrator command
SSL connection has been closed unexpectedly
 'SELECT "user".id AS user_id, "user".email AS user_email, "user".password AS user_password, "user".active AS user_active, "user".confirmed_at AS user_confirmed_at, "user".daily_audit_email AS user_daily_audit_email, "user".change_reports AS user_change_reports, "user".last_login_at AS user_last_login_at, "user".current_login_at AS user_current_login_at, "user".login_count AS user_login_count, "user".last_login_ip AS user_last_login_ip, "user".current_login_ip AS user_current_login_ip, "user".role AS user_role \nFROM "user" \nWHERE "user".id = %(id_1)s \n LIMIT %(param_1)s' {'id_1': u'1', 'param_1': 1}                                                                                              

```

EDIT: 9/15 by mikegrima - added markdown for readability

**Comment by mikegrima** [comment]
You are having SQLAlchemy errors.  It looks as though it is having database connectivity issues.

I'm not familiar with this specific error:
```
OperationalError: (OperationalError) terminating connection due to administrator command
SSL connection has been closed unexpectedly
```

Were you able to run the `monkey` commands before successfully?  What happens when you run: `monkey amazon_accounts`?

**Comment by andrewpsp** [comment]
After going through multiple issues and referencing my error message 

After following the build instructions with precision, and creating IAM settings with perfection. When performing the monkey db upgrade this the error I'm facing consistently. The goal being to connect to a Postgres RDS instance. I sure many have faced this particular issue in the past. Note - My UI was not populating any relevant information.

``` conn = _connect(dsn, connection_factory=connection_factory, async=async) sqlalchemy.exc.OperationalError: (OperationalError) could not connect to server: Connection timed out Is the server running on host "omitted" (omitted) and accepting TCP/IP connections on port 5432? ```

even after digging through the migrations/env.py trying to print feedback per for the value of SQLALCHEMY_DATABASE_URI  with no success https://github.com/Netflix/security_monkey/issues

Step taken  temporarily open SG settings from the instance and from the DB. 
Hardcode values in the default-config.py and env-config/config.py with completely no progress. 


After banging my head it running across this last issue where I simply service ufw stop and here is the the tail after stoping the ufw and running monkey db upgrade

```
scheduler.py", line 208, in setup_scheduler\n    accounts = Account.query.filter(Account.third_party == False).filter(Account.active == True).all()  # noqa\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2279, in all\n    return list(self)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2391, in __iter__\n    return self._execute_and_instances(context)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2406, in _execute_and_instances\n    result = conn.execute(querycontext.statement, self._params)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 717, in execute\n    return meth(self, multiparams, params)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/sql/elements.py", line 317, in _execute_on_connection\n    return connection._execute_clauseelement(self, multiparams, params)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 814, in _execute_clauseelement\n    compiled_sql, distilled_params\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 927, in _execute_context\n    context)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1076, in _handle_dbapi_exception\n    exc_info\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 185, in raise_from_cause\n    reraise(type(exception), exception, tb=exc_tb)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 920, in _execute_context\n    context)\n  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line 425, in do_execute\n    cursor.execute(statement, parameters)\nProgrammingError: (ProgrammingError) relation "account" does not exist\nLINE 2: FROM account \n             ^\n \'SELECT account.id AS account_id, account.active AS account_active, account.third_party AS account_third_party, account.name AS account_name, account.notes AS account_notes, account.identifier AS account_identifier, account.account_type_id AS account_account_type_id \\nFROM account \\nWHERE account.third_party = false AND account.active = true\' {}\n', 'account_id': None, 'region': None, 'occurred': datetime.datetime(2017, 10, 18, 18, 59, 13, 425950), 'tech_id': None, 'source': 'scheduler', 'ttl': datetime.datetime(2017, 10, 28, 18, 59, 13, 425983), 'item_id': None, 'message': '(ProgrammingError) relation "account" does not exist\nLINE 2: FROM account \n             ^\n \'SELECT account.id AS account_id, account.active AS account_active, account.third_party AS account_third_party, account.name AS account_name, account.notes AS account_notes, account.identifier AS account_identifier, account.account_type_id AS account_account_type_id \\nFROM account \\nWHERE account.third_party = false AND account.active = true\' {}', 'type': 'ProgrammingError'}
2017-10-18 18:59:14,019 INFO: Scheduler started [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:574]
2017-10-18 18:59:14,019 INFO: Scheduler has been shut down [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:606]
2017-10-18 18:59:14,120 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-10-18 18:59:15,979 INFO: Started thread pool with 25 core threads and 30 maximum threads [in build/bdist.linux-x86_64/egg/apscheduler/threadpool.py:52]
2017-10-18 18:59:16,267 INFO: Adding job tentatively -- it will be properly scheduled when the scheduler starts [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:287]
2017-10-18 18:59:16,267 INFO: Added job "_clear_old_exceptions (trigger: cron[hour='3', minute='0'], next run at: 2017-10-19 03:00:00)" to job store "default" [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:262]
2017-10-18 18:59:16,268 INFO: Scheduler started [in build/bdist.linux-x86_64/egg/apscheduler/scheduler.py:574]
```



To summarize stoping the ufw service resolved the issue.

**Comment by monkeysecurity** [comment]
Woa.  Are you using any custom ufw configuration or just stock Ubuntu?

**Comment by monkeysecurity** [comment]
Closing issue as a workaround has been found.  Not sure that SM can do anything for this issue.  First I've heard of it.

**Comment by andrewpsp** [comment]
@monkeysecurity  my instances are behind a VPN and VPC so the ubuntu instance ufw service was completely stopped to resolve the issue.
